chalk-ir/src/cast.rs

```diff
@@ -225,12 +208,6 @@ where
     }
 }
 
-impl<I: Interner> CastTo<ProgramClause<I>> for ProgramClauseImplication<I> {
```

Comment:
So I think *this* impl was completely unused...

Reply:
Yes, this and the one underneath (for `Binders<ProgramClauseImplication<I>>`